CMakeLists.txt

```
@@ -90,10 +69,35 @@ if(NOT WIN32 OR CMAKE_COMPILER_IS_GNUCXX)
 endif()
 
 if(BUILD_PYTHON)
-    set(SIP_EXTRA_FILES_DEPEND python/Types.sip python/MeshData.sip python/SceneNode.sip python/Scene.sip)
-    #set(SIP_EXTRA_SOURCE_FILES python/Types.cpp)
-    set(SIP_EXTRA_OPTIONS -g -n PyQt5.sip) # -g means always release the GIL before calling C++ methods. -n PyQt5.sip is required to not get the PyCapsule error
-    add_sip_python_module(Savitar python/ThreeMFParser.sip Savitar)
+    list(APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake)
+
+    if(NOT DEFINED Python_VERSION)
+        set(Python_VERSION
+                3.10
+                CACHE STRING "Python Version" FORCE)
+        message(STATUS "Setting Python version to ${Python_VERSION}. Set Python_VERSION if you want to compile against an other version.")
+    endif()
+    if(APPLE)
+        set(Python_FIND_FRAMEWORK NEVER)
```

```suggestion
        set(Python_FIND_FRAMEWORK NEVER)
```

This could use a bit of documentation as to why this should be set to NEVER in the case of osx builds (and yes, i know, it was also like this in the old code)

---

Because MacOS Framework python is the root of all evil ;-) and we don't want to accidentally link against it. I think the variable name and value say's it all. btw I could add a one liner of doc in there. But I don't think it will add much additional information